src/algebra/ordered_monoid.lean

```diff
@@ -1117,6 +1117,11 @@ by simpa [add_comm] using @with_bot.add_lt_add_iff_left _ _ a b c
 
 end ordered_cancel_add_comm_monoid
 
+/-- an `ordered_cancel_add_comm_monoid` with one-sided 'subtraction'
+in the sense that if `a ≤ b`, there is some `c` for which `a + c = b` -/
+class has_exists_add_of_le (α : Type u) [ordered_cancel_add_comm_monoid α] :=
```

Comment:
Could relax the typeclass to:
```suggestion
class has_exists_add_of_le (α : Type u) [ordered_add_comm_monoid α] :=
```

Reply:
> Could relax the typeclass to:

I think cancellation is needed for the map to be a bijection; if we have ``a + d = b + d`` and ``a \ne b`` then the map (+d) doesn't give an injection on the closed interval ``[a,b]``.